tree/ntuple/v7/src/RNTupleMerger.cxx

```diff
@@ -22,10 +22,44 @@
 
 Long64_t ROOT::Experimental::RNTuple::Merge(TCollection *inputs, TFileMergeInfo *mergeInfo)
 {
-   if (inputs == nullptr || mergeInfo == nullptr) {
-      return -1;
+   // Check the inputs
+   if (!inputs || inputs->GetEntries() < 3 || !mergeInfo) return -1;
+
+   // Input parameters
+   std::string ntupleName;
+   std::unique_ptr<Detail::RPageSink> destination;
+   std::vector<std::unique_ptr<Detail::RPageSource>> sources;
+   std::vector<Detail::RPageSource *> sourcePtrs;
+
+   // Parse the inputs
+   Long64_t counter {0};
+   TIter itr(inputs);
+   while(const auto& pitr = itr()) {
```

Comment:
We may want to iterate the inputs explicitly (without the while loop).

Reply:
Thanks Jakob. We now dropped this counter approach. The first two entries are processed explicitly. The input files are still being processed through a while loop but that seems like the cleanest approach.